services/operations.go

```diff
@@ -533,165 +646,205 @@ func (m *transactionToOperationMapper) EmitFeeOps() {
+		return fmt.Errorf("malformed body: %w", err)
+	}
+
+	opIndex := int64(len(m.ops))
+	m.ops = append(m.ops,
+		&types.Operation{
+			OperationIdentifier: &types.OperationIdentifier{
+				Index: opIndex,
+			},
+			Type:   OpTransfer,
+			Status: m.status,
+			Account: &types.AccountIdentifier{
+				Address: m.txSignerAddress,
+			},
+			Amount: &types.Amount{
+				Value:    "-" + body.Amount.String(),
+				Currency: OasisCurrency,
 			},
-			&types.Operation{
-				OperationIdentifier: &types.OperationIdentifier{
-					Index: opIndex + 1,
+		},
+		&types.Operation{
+			OperationIdentifier: &types.OperationIdentifier{
+				Index: opIndex + 1,
+			},
+			Type:   OpTransfer,
+			Status: m.status,
+			Account: &types.AccountIdentifier{
+				Address: StringFromAddress(body.Account),
+				SubAccount: &types.SubAccountIdentifier{
+					Address: SubAccountEscrow,
 				},
-				Type:   OpReclaimEscrow,
-				Status: m.status,
-				Account: &types.AccountIdentifier{
-					Address: StringFromAddress(body.Account),
-					SubAccount: &types.SubAccountIdentifier{
-						Address: SubAccountEscrow,
-					},
+			},
+			Amount: &types.Amount{
+				Value:    body.Amount.String(),
+				Currency: OasisCurrency,
+			},
+			RelatedOperations: []*types.OperationIdentifier{
+				{
+					Index: opIndex,
 				},
-				Metadata: map[string]interface{}{
-					ReclaimEscrowSharesKey: body.Shares.String(),
+			},
+		},
+	)
+
+	return nil
+}
+
+// emitReclaimEscrowOps emits the required operations for the reclaim escrow
+// transaction.
+func (m *transactionToOperationMapper) emitReclaimEscrowOps() error {
+	var body staking.ReclaimEscrow
+	if err := cbor.Unmarshal(m.tx.Body, &body); err != nil {
+		return fmt.Errorf("malformed body: %w", err)
+	}
+
+	opIndex := int64(len(m.ops))
+	m.ops = append(m.ops,
+		&types.Operation{
+			OperationIdentifier: &types.OperationIdentifier{
+				Index: opIndex,
+			},
+			Type:   OpReclaimEscrow,
+			Status: m.status,
+			Account: &types.AccountIdentifier{
+				Address: m.txSignerAddress,
+			},
+		},
+		&types.Operation{
+			OperationIdentifier: &types.OperationIdentifier{
+				Index: opIndex + 1,
+			},
+			Type:   OpReclaimEscrow,
+			Status: m.status,
+			Account: &types.AccountIdentifier{
+				Address: StringFromAddress(body.Account),
+				SubAccount: &types.SubAccountIdentifier{
+					Address: SubAccountEscrow,
 				},
-				RelatedOperations: []*types.OperationIdentifier{
-					&types.OperationIdentifier{
-						Index: opIndex,
-					},
+			},
+			Metadata: map[string]interface{}{
+				ReclaimEscrowSharesKey: body.Shares.String(),
+			},
+			RelatedOperations: []*types.OperationIdentifier{
+				{
+					Index: opIndex,
 				},
 			},
-		)
+		},
+	)
+
+	return nil
+}
+
+// EmitTxOps emits the required transaction-specific operations.
+func (m *transactionToOperationMapper) EmitTxOps() error {
+	switch m.tx.Method {
+	case staking.MethodTransfer:
+		if err := m.emitTransferOps(); err != nil {
+			return err
+		}
```

Comment:
Why not just `return m.emitTransferOps()` and similar below?

Reply:
Indeed. Done.